InterfacePython/TasmanianAddons.py

```diff
@@ -402,3 +409,46 @@ def loadUnstructuredDataL2(points, model_data, tolerance, grid):
     num_data = points.shape[0]
     pLibCTSG.tsgLoadUnstructuredDataL2(np.ctypeslib.as_ctypes(points.reshape((points.size,))), num_data,
                                        np.ctypeslib.as_ctypes(model_data.reshape((model_data.size,))), tolerance, grid.pGrid)
+
+def createExoticQuadratureFromGrid(level, shift, ref_grid, description, is_symmetric = False):
+    '''
+    Calls TasGrid::getExoticQuadrature() from a one dimensional interpolant/surrogate of the weight function, and output a
+    python CustomTabulated object.
+    See the C++ reference for more information.
+
+    level:        positive integer representing the level of the exotic quadrature grid.
+    shift:        double where [weight_function(x) + shift] is nonegative for every x in the domain of the weight function.
+    ref_grid:     Python TasmanianSparseGrid object that represents the one dimensional surrogate/interpolant of the weight function.
+    description:  string describing the Exotic quadrature instance.
+    is_symmetric: (optional) boolean that should be set to True if the weight function is symmetric.
+
+    output:       a Python CustomTabulated object.
+    '''
+    if not hasattr(ref_grid, "TasmanianSparseGridObject"):
+        raise TasmanianInputError("ref_grid", "ERROR: ref_grid must be an instance of TasmanianSparseGrid")
+    effective_description = bytes(description, encoding='utf8') if sys.version_info.major == 3 else description
+    ct = TasmanianSG.CustomTabulated()
+    ct.pCustomTabulated = pLibCTSG.tsgCreateExoticQuadratureFromGrid(c_int(level), c_double(shift), c_void_p(ref_grid.pGrid),
```

Comment:
This is leaking memory.

Both `TasmanianSG.CustomTabulated()` and `pLibCTSG.tsgCreateExoticQuadratureFromGrid()` are calling `new` on C++ pointers, but `ct` has only one destructor and only one C++ `delete` will be issued.

Reply:
Changed the C functions to modify ``ct.pCustomTabulated`` directly rather than having them return a new object on the heap.